Sources/Host/Darwin/PTrace.cpp

```diff
@@ -57,16 +58,8 @@ ErrorCode PTrace::kill(ProcessThreadId const &ptid, int signal) {
 ErrorCode PTrace::readString(ProcessThreadId const &ptid,
                              Address const &address, std::string &str,
                              size_t length, size_t *count) {
-  char buf[length];
-  ErrorCode err = readMemory(ptid, address, buf, length, count);
-  if (err != kSuccess)
-    return err;
-
-  if (strnlen(buf, length) == length)
-    return kErrorNameTooLong;
-
-  str = std::string(buf);
-  return kSuccess;
+  DS2BUG("not implemented");
+  return kErrorUnsupported;
```

Comment:
`DS2BUG` is marked `__attribute__((__noreturn__))` so you don't need the `return kErrorUnsupported;` here.